### PR TITLE
Regenerate build context using ansible-builder 0.4.0

### PR DESCRIPTION
**Pull request by shanemcd**

### CODE REVIEW

File: requirements.yml [hunk 1]
```diff
@@ -12,7 +12,8 @@ collections:
     version: ee_req_install
     type: git
   - name: community.vmware  # has requirements.txt, but may add pyvcloud
-  - name: ovirt.ovirt
+  - name: https://github.com/shanemcd/ovirt-ansible-collection
+    type: git
```
Comment (pabelanger) [inline requirements.yml:16]: So, for this. I will update our build jobs, to allow us to use depends-on for this too.

The trick is, we need to either allow for git contents inside galaxy layer, like we did with ZUUL_SIBLINGS, or pre-compiled tarballs.

Either way, soon we won't have to do this for CI. Which I am excited for.

File: bindep_combined.txt [hunk 2]
```diff
@@ -4,11 +4,14 @@ libyaml-dev [test platform:dpkg]  # from collection ansible.netcommon
...
+gcc [platform:rpm compile]  # from collection ovirt.ovirt
+libcurl-devel [platform:rpm compile]  # from collection ovirt.ovirt
+libxml2-devel [platform:rpm compile]  # from collection ovirt.ovirt
 python3-pycurl [platform:rpm]  # from collection ovirt.ovirt
 python3-netaddr [platform:rpm]  # from collection ovirt.ovirt
 python3-jmespath [platform:rpm]  # from collection ovirt.ovirt
-python3-passlib [platform:rpm]  # from collection ovirt.ovirt
+python3-passlib [platform:rpm epel]  # from collection ovirt.ovirt
 qemu-img [platform:rpm]  # from collection ovirt.ovirt
 python3-devel [platform:rpm]  # from collection user
```
Comment (pabelanger) [inline bindep_combined.txt]: this comes from tools/bindep.txt, we should move it to compile or just drop it.

Comment (pabelanger) [reply bindep_combined.txt]: python3-devel [platform:rpm]

File: Containerfile [hunk 3]
```diff
@@ -7,13 +7,18 @@ RUN ansible-galaxy collection install -r /build/requirements.yml --collections-p
...
 
+FROM quay.io/ansible/python-builder:latest as builder
+
+ADD requirements_combined.txt /tmp/src/requirements.txt
+ADD bindep_combined.txt /tmp/src/bindep.txt
+RUN assemble
+
 FROM quay.io/ansible/ansible-runner:devel
+
 RUN pip3 install --upgrade pip setuptools
```
Comment (pabelanger) [inline Containerfile]: this is redundant now, but we can fix later.

**Review by pabelanger — APPROVED** [review]

File: Containerfile (hunk 3 above)
Comment (shanemcd) [reply Containerfile]: +1, this will require a change in builder itself.

Comment (shanemcd) [reply Containerfile]: oops. nvm. it's in the definition!

Comment (pabelanger) [reply Containerfile]: from tools/execution-environment.yaml prepend.

Comment (shanemcd) [reply Containerfile]: addressed in latest commit.

**Pull request merged**